### PR TITLE
Untainting exim config for sieve

### PR DESCRIPTION
**Pull request by henri-hulski**
In the `dovecot_virtual_delivery` transport I've replaced the command with
```
  command = /usr/lib/dovecot/dovecot-lda -e -d ${extract{1}{:}{${lookup{$local_part}lsearch{/etc/exim4/domains/${lookup{$domain}dsearch{/etc/exim4/domains/}}/accounts}}}}@${lookup{$domain}dsearch{/etc/exim4/domains/}}
```
So I replaced the username (-d) with an un-tainted version,
removed the envelop_sender (-f) as the default for message_prefix is the mbox string "From " and Dovecot will use it to identify the sender and
removed the address (-a) which defaults to the username. 
Works for me. I hope I didn't missed anything.

### CODE REVIEW

File: install/hst-install-debian.sh [hunk 1]
```diff
@@ -1876,7 +1876,7 @@ if [ "$sieve" = 'yes' ]; then
...
-	sed -i "s/address_pipe:/dovecot_virtual_delivery:\n  driver = pipe\n  command = \/usr\/lib\/dovecot\/dovecot-lda -e -d \$local_part@\$domain -f \$sender_address -a \$original_local_part@\$original_domain\n  delivery_date_add\n  envelope_to_add\n  return_path_add\n  log_output = true\n  log_defer_output = true\n  user = \${extract{2}{:}{\${lookup{\$local_part}lsearch{\/etc\/exim4\/domains\/\${lookup{\$domain}dsearch{\/etc\/exim4\/domains\/}}\/passwd}}}}\n group = mail\n  return_output\n\naddress_pipe:/g" /etc/exim4/exim4.conf.template
+	sed -i "s/address_pipe:/dovecot_virtual_delivery:\n  driver = pipe\n  command = \/usr\/lib\/dovecot\/dovecot-lda -e -d \${extract{1}{:}{\${lookup{\$local_part}lsearch{/etc/exim4/domains/\${lookup{\$domain}dsearch{/etc/exim4/domains/}}/accounts}}}}@\${lookup{\$domain}dsearch{/etc/exim4/domains/}}\n  delivery_date_add\n  envelope_to_add\n  return_path_add\n  log_output = true\n  log_defer_output = true\n  user = \${extract{2}{:}{\${lookup{\$local_part}lsearch{\/etc\/exim4\/domains\/\${lookup{\$domain}dsearch{\/etc\/exim4\/domains\/}}\/passwd}}}}\n  group = mail\n  return_output\n\naddress_pipe:/g" /etc/exim4/exim4.conf.template
```
Comment (jaapmarcus) [inline install/hst-install-debian.sh]: Backward slashes needs also to be escaped

File: install/hst-install-ubuntu.sh [hunk 2]
```diff
@@ -1927,7 +1927,7 @@ if [ "$sieve" = 'yes' ]; then
...
-	sed -i "s/address_pipe:/dovecot_virtual_delivery:\n  driver = pipe\n  command = \/usr\/lib\/dovecot\/dovecot-lda -e -d \$local_part@\$domain -f \$sender_address -a \$original_local_part@\$original_domain\n  delivery_date_add\n  envelope_to_add\n  return_path_add\n  log_output = true\n  log_defer_output = true\n  user = \${extract{2}{:}{\${lookup{\$local_part}lsearch{\/etc\/exim4\/domains\/\${lookup{\$domain}dsearch{\/etc\/exim4\/domains\/}}\/passwd}}}}\n  group = mail\n  return_output\n\naddress_pipe:/g" /etc/exim4/exim4.conf.template
+	sed -i "s/address_pipe:/dovecot_virtual_delivery:\n  driver = pipe\n  command = \/usr\/lib\/dovecot\/dovecot-lda -e -d \${extract{1}{:}{\${lookup{\$local_part}lsearch{/etc/exim4/domains/\${lookup{\$domain}dsearch{/etc/exim4/domains/}}/accounts}}}}@\${lookup{\$domain}dsearch{/etc/exim4/domains/}}\n  delivery_date_add\n  envelope_to_add\n  return_path_add\n  log_output = true\n  log_defer_output = true\n  user = \${extract{2}{:}{\${lookup{\$local_part}lsearch{\/etc\/exim4\/domains\/\${lookup{\$domain}dsearch{\/etc\/exim4\/domains\/}}\/passwd}}}}\n  group = mail\n  return_output\n\naddress_pipe:/g" /etc/exim4/exim4.conf.template
```
Comment (jaapmarcus) [inline install/hst-install-ubuntu.sh]: Same here

File: install/upgrade/manual/install_sieve.sh [hunk 3]
```diff
@@ -113,7 +113,7 @@ else
...
-		sed -i "s/dovecot_virtual_delivery:\n  driver = pipe\n  command = \/usr\/lib\/dovecot\/dovecot-lda -e -d \$local_part@\$domain -f \$sender_address -a \$original_local_part@\$original_domain\n  delivery_date_add\n  envelope_to_add\n  return_path_add\n  log_output = true\n  log_defer_output = true\n  user = \${extract{2}{:}{\${lookup{\$local_part}lsearch{\/etc\/exim4\/domains\/\${lookup{\$domain}dsearch{\/etc\/exim4\/domains\/}}\/passwd}}}}\n  group = mail\n  return_output\n//g" /etc/exim4/exim4.conf.template
+		sed -i "s/dovecot_virtual_delivery:\n  driver = pipe\n  command = \/usr\/lib\/dovecot\/dovecot-lda -e -d \${extract{1}{:}{\${lookup{\$local_part}lsearch{/etc/exim4/domains/\${lookup{\$domain}dsearch{/etc/exim4/domains/}}/accounts}}}}@\${lookup{\$domain}dsearch{/etc/exim4/domains/}}\n  delivery_date_add\n  envelope_to_add\n  return_path_add\n  log_output = true\n  log_defer_output = true\n  user = \${extract{2}{:}{\${lookup{\$local_part}lsearch{\/etc\/exim4\/domains\/\${lookup{\$domain}dsearch{\/etc\/exim4\/domains\/}}\/passwd}}}}\n  group = mail\n  return_output\n//g" /etc/exim4/exim4.conf.template
```
Comment (jaapmarcus) [inline install/upgrade/manual/install_sieve.sh]: And here

**Review by jaapmarcus — CHANGES_REQUESTED** [review]

File: install/hst-install-debian.sh (hunk 1 above)
Comment (henri-hulski) [reply install/hst-install-debian.sh]: There are no back slashes in the command?

File: install/hst-install-ubuntu.sh (hunk 2 above)
Comment (henri-hulski) [reply install/hst-install-ubuntu.sh]: I escaped all slashes with backslashes and removed an unintended commit.

**Comment by jaapmarcus** [comment]
We need to write / test it but Debain 12 is 6 months aways probally work after 1.7 release

**Comment by jaapmarcus** [comment]
Thanks have made the changes in 
https://github.com/hestiacp/hestiacp/pull/3661

**Pull request closed without merging**